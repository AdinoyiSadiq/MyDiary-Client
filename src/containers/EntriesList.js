import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getEntriesList } from '../actions/entriesListActions';
import EmptyListModal from '../components/EmptyListModal';

class EntriesList extends Component {
  componentWillMount() {
    const { getEntriesList } = this.props;
    getEntriesList();
  }

  renderEntriesList(entriesList) {
    const entries = entriesList.map((entry) => {
      const date = new Date(parseInt(entry.created));
      return (
        <article key={entry.id} className="entry">
          <div className="content">
            <Link to={`/main/entry/${entry.id}`}>
              <h3 className="title">{entry.title}</h3>
              <p className="date">{date.toString().substring(0, 15)}</p>
              <p className="entryContent">{(entry.content).substring(0, 100)}</p>
            </Link>
            <div className="actions">
              <div title="Edit"><i className="far fa-edit"></i></div>
              <div title="Delete"><i className="far fa-trash-alt" id="delete"></i></div>
            </div>
          </div>
        </article>
      );
    });
    return (
      <section className="entryList" id="list">
        {entries}
      </section>
    );
  }

  render() {
    const { entries } = this.props;
    if (entries) {
      return (
        <main className="clearfix">
          {(entries.length > 1) ? this.renderEntriesList(entries) : <EmptyListModal />}
        </main>
      );
    }
    return <div>Loading...</div>
  }
};

function mapStateToProps(state) {
  return { entries: state.entriesList.entries };
}

export default connect(mapStateToProps, { getEntriesList })(EntriesList);