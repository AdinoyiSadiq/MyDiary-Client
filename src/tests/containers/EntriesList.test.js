import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import EntriesList from '../../containers/EntriesList';
import Root from '../../root';

let noEntries;
let withEntries;

const mockState = { 
  entriesList: [{
    created: 1543484351357
  }],
};


beforeEach(() => {
  noEntries = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <EntriesList />
      </MemoryRouter>
    </Root>,
  );

  withEntries = mount(
    <Root initialState={mockState}>
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <EntriesList />
      </MemoryRouter>
    </Root>,
  );
});

afterEach(() => noEntries.unmount());
afterEach(() => withEntries.unmount());

describe('EntriesList UI', () => {
  describe('render features', () => {
    test('container should render as expected when there are no entries', () => {
      const tree = toJson(noEntries);
      expect(tree).toMatchSnapshot();
    });

    test('container should render as expected when there are entries', () => {
      const tree = toJson(withEntries);
      expect(tree).toMatchSnapshot();
    });
  });
});