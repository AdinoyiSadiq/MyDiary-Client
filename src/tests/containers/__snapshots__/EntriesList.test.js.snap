// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntriesList UI render features container should render as expected when there are entries 1`] = `
<_default
  initialState={
    Object {
      "entriesList": Array [
        Object {
          "created": 1543484351357,
        },
      ],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(EntriesList)>
          <EntriesList
            deleteEntry={[Function]}
            entries={[Function]}
            getEntriesList={[Function]}
          >
            <main
              className="clearfix"
            >
              <DeleteEntryModal
                deleteEntry={[Function]}
                display={false}
                hideDeleteModal={[Function]}
              >
                <section
                  className="authContent deleteModal null"
                >
                  <div
                    className="title"
                  >
                    Do you want to delete this entry?
                  </div>
                  <hr />
                  <div
                    className="clearfix"
                  >
                    <button
                      id="deleteButton"
                      onClick={[Function]}
                    >
                      Yes
                    </button>
                    <button
                      id="cancelButton"
                      onClick={[Function]}
                    >
                      No
                    </button>
                  </div>
                </section>
              </DeleteEntryModal>
              <EmptyListModal>
                <section
                  className="authContent emptyList"
                >
                  <div
                    className="title"
                  >
                    You don't seem to have created an entry
                  </div>
                  <hr />
                  <form>
                    <Link
                      replace={false}
                      to="/main/addEntry"
                    >
                      <a
                        href="/main/addEntry"
                        onClick={[Function]}
                      >
                        <button>
                          Create an Entry
                        </button>
                      </a>
                    </Link>
                  </form>
                </section>
              </EmptyListModal>
            </main>
          </EntriesList>
        </Connect(EntriesList)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;

exports[`EntriesList UI render features container should render as expected when there are no entries 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(EntriesList)>
          <EntriesList
            deleteEntry={[Function]}
            getEntriesList={[Function]}
          >
            <div>
              Loading...
            </div>
          </EntriesList>
        </Connect(EntriesList)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
